draft/spec/index.html

```diff
@@ -975,30 +975,30 @@ <h2>Root Structure</h2>
           <a href="#storage-root-extensions">storage root extensions</a>. An OCFL validator
           <span id="E087">MUST</span> ignore any files in the storage root it does not understand.
         </p>
+        <p>
+          An OCFL Storage Root <span id="E088">MUST NOT</span> contain directories or sub-directories other than as
+          a directory hierarchy used to store OCFL Objects or for <a href="#storage-root-extensions">storage root
+          extensions</a>. The directory hierarchy used to store OCFL Objects <span id="E072">MUST NOT</span> contain
+          files that are not part of an OCFL Object. Empty directories <span id="E073">MUST NOT</span> appear under
+          a storage root.
+        </p>
         <p>
           An OCFL Storage Root MAY contain a JSON file named <code>ocfl_layout.json</code> to describe the arrangement
           of directories and OCFL objects under the storage root. If present, this JSON document
           <span id="E070">MUST</span> include the following two keys in the root JSON object:
         </p>
         <ul>
           <li>
-            <code>key</code> - A key identifying the precise arrangement of directories and OCFL objects under the
-            storage root, i.e. how OCFL object identifiers are mapped to directory hierarchies. The value of this key
-            is not defined in the OCFL specification, but <span id="E071">MUST</span> correspond to a value given in an
-            <a>Extension</a>.
+            <code>extension</code> - An extension name that identifies an arrangement of directories and OCFL objects
+            under the storage root, i.e. how OCFL object identifiers are mapped to directory hierarchies. The
+            value of the <code>extension</code> key <span id="E071">MUST</span> be the registered <a>extension</a>
```

Comment:
There is still room for misinterpretation of what is meant by "registered extension name". I would suggest we refer to the proposed `Extension ID` seen in this pull-request: https://github.com/OCFL/extensions/pull/16/files#diff-77315547024f9923f13fb0a37b4ecf3aR3

Reply:
Or maybe we should fix that PR to say extension name? I'm loathe to call something that isn't a URI or similar an id

Reply:
I don't mind updating that PR. I added that line speculatively.

Reply:
cf. "Digest Algorithm Name" also in https://ocfl.github.io/extensions/0001-digest-algorithms.html

Reply:
Using a URI would be nice. 
It is important that the language here be very specific, unambiguous.

Reply:
https://github.com/OCFL/extensions/pull/13 also has updates around extension names.